src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/extension/env/EnvExtension.groovy

```diff
@@ -93,6 +106,22 @@ class EnvExtension extends AbstractGlobalExtension implements SpringContextListe
             throw new RuntimeException("Provided profile '$profile' is unknown. Select one of the following profiles:" +
                     " hardware, virtual")
         }
+        if((profile == "hardware" && enable) || (profile == "virtual" && !isTopologyRebuildRequired && enable)) {
+            log.info("Starting topology-related HC: lab_id=" + topology.labId)
+            Wrappers.wait(30) {
+                withPool {
+                    [allSwitchesAreActive,
+                     allLinksAreActive,
+                     noFlowsLeft,
+                     noLinkPropertiesLeft,
+                     linksBandwidthAndSpeedMatch,
+                     noExcessRulesMeters,
+                     allSwitchesConnectedToExpectedRegion,
+                     switchesConfigurationIsCorrect].eachParallel { it(topology) }
+                }
+            }
+            log.info("Topology-related HC passed: lad_id=" + topology.labId + " time: " + System.currentTimeSeconds())
```

Comment:
Misprint: should be `lab_id`

Also, either add start time to the line 110 or remove it from this line. Otherwise logging this doesn't make sense.
As an option, you may wrap method into `Wrappers.benchmark()` (though, you'll have to add logger parameter to it to output into console instead of logstash).

Reply:
sorry, it was my debug time, I don't want to use TOPOLOGY_DISCOVERING_TIME, added the new constant for that HEALTH_CHECK_TIME.